test/unit/carousel.ts

```diff
@@ -0,0 +1,414 @@
+        { content: 'test7' },
+        { content: 'test8' },
+        { content: 'test9' },
+      ];
+    });
+
+    describe('onMount()', () => {
+      it('should add event listener for window resize', () => {
+        const addEventListener = spy();
+        stub(utils, 'WINDOW').returns({ addEventListener });
+
+        carousel.onMount();
+
+        expect(addEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('onUpdate()', () => {
+      it('should add cloned items to state', () => {
+        const clone = carousel.cloneItems = spy();
+
+        carousel.onUpdate();
+
+        expect(clone).to.be.called;
+      });
+    });
+
+    describe('onUnmount()', () => {
+      it('should add event listener for window resize', () => {
+        const removeEventListener = spy();
+        stub(utils, 'WINDOW').returns({ removeEventListener });
+
+        carousel.onUnmount();
+
+        expect(removeEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('moveNext', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.moveNext();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(1);
+      });
+    });
+
+    describe('movePrevious', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.movePrevious();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(-1);
+      });
+    });
+
+    describe('isSwipeToNext()', () => {
+      it('should move next when swiping left with an angle between 315 and 360', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, 4);
+        expect(direction).to.be.true;
+      });
+
+      it('should move next when swiping left with an angle between 0 and 45', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, -10);
+        expect(direction).to.be.true;
+      });
+
+      it('should move previous when swiping right with an angle between 135 and 225', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 25, 10);
+        expect(direction).to.be.false;
+      });
+
+      it('should not return anything if angle is between 45 to 135 or between225 and 315', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 0, 100);
+        expect(direction).to.be.null;
+      });
+    });
+
+    describe('onTouchStart()', () => {
+      it('should add listener', () => {
+        const event = {
+          touches: [{ pageX: 0, pageY: 5 }],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        carousel.set = () => {
```

Comment:
we should be setting `carousel.set` to a `spy()` and then checking the call params of the spy rather than mocking out the behaviour of `set()` then verifying it